### Варіант № 3
### Завдання № 57(а)
Дано дійсне число $a$. Визначити $f(a)$, якщо
$f(x)=\begin{cases}x^2 &\text{при } -2 \le x<2 \\
4 &\text{в іншому випадку } \end{cases}$

In [10]:
try:
    a = float(input("Enter number a: "))
    print("f(a) = ", a ** 2) if -2 <= a < 2 else print("f(a) = 4")
except ValueError:
    print("Input error!")

f(a) =  0.0


# Examples

### Example 1
Enter number a: 1.5
f(a) =  2.25

### Example 2
Enter number a: -5
f(a) =  4

### Example 3
Enter number a: 0
f(a) =  0.0

### Завдання № 35
Дано дійсні числа $x, y, z$. Визначити
1. $max(x+y+z, xyz);$
2. $min^2(x+y+z/2, xyz)+1$.

In [13]:
try:
    x = float(input("Enter x = "))
    y = float(input("Enter y = "))
    z = float(input("Enter z = "))
    print("a)", max(x+y+z, x*y*z), "\nb)", (min(x+y+z/2, x*y*z))**2+1)
except ValueError:
    print("Input error!")

a) -0.0 
b) 10.0


# Examples

### Example 1
Enter x = 1
Enter y = 1
Enter z = 1
a) 3.0
b) 2.0

### Example 2
Enter x = 12
Enter y = -22
Enter z = 0.25
a) -9.75
b) 4357.0

### Example 3
Enter x = -4
Enter y = 0
Enter z = 2
a) 0.0
b) 10.0

### Завдання № 49
Дано дійсне число $h$. Визначити чи має рівняння $ax^2+bx+c=0$ дійсні корені,
якщо
1. $a=\sqrt\frac{|\sin 8h|+17}{(1-\sin 4h \cos(h^2+18))^2}$
2. $b=1-\sqrt\frac{3}{3+|\tan ah^2-\sin ah|}$
3. $c=ah^2\sin bh+bh^3\cos ah$.

Якщо дійсні корені існують, то знайти їх. В протилежному випадку відповіддю
має слугувати повідомлення, що дійсних коренів немає.

In [18]:
import math

try:
    h = float(input("Enter number h: "))
    a = math.sqrt((abs(math.sin(8*h))+17)/((1-math.sin(4*h)*math.cos(h**2+18))**2))
    b = 1-math.sqrt(3/(3+abs(math.tan(a*h**2)-math.sin(a*h))))
    c = a*h**2*math.sin(b*h)+b+h**3+math.cos(a*h)
    d = b**2-4*a*c
    if d > 0:
        print("Your roots: \nx1 =", (-b+math.sqrt(d))/2*a,
              "\nx2 =", (-b-math.sqrt(d))/2*a)
    elif d == 0:
        print("Your root: \nx =", -b/(2*a))
    else:
        print("No roots")
except ValueError:
    print("Input error!")

Your roots: 
x1 = 218.06739364436382 
x2 = -218.27010959267605


# Examples

### Example 1
Enter number h: -5
Your roots:
x1 = 63.67828606407461
x2 = -64.3020493283624

### Example 2
Enter number h: 0
No roots

### Example 3
Enter number h: -7
Your roots:
x1 = 218.06739364436382
x2 = -218.27010959267605
